src/ex1/script.js

```diff
@@ -0,0 +1,284 @@
+  const todoLi = document.createElement("li");
+  todoLi.classList.add("todo");
+  todoLi.innerText = todoInput.value;
+
+  //Add todo to local storage
+  saveLocalTodos(todoInput.value);
+
+  //trash button
+  const trashButton = document.createElement("button");
+  trashButton.innerHTML = '<i  class="fas fa-trash"></i>';
+  trashButton.classList.add("trash-btn");
+  todoLi.appendChild(trashButton);
+  //append to list
+  todoList.appendChild(todoLi);
+
+  const listChildrenCount = todoList.childElementCount;
+  listStyling(listChildrenCount);
+
+  //Clear Todo Input value
+  todoInput.value = "";
+  inputBox.classList.add("inactive");
+}
+
+//add todo when pressing enter key
+function todoEnter(event) {
+  if (event.keyCode == ENTER_KEY) {
+    if (!todoInput.value.trim()) {
+      return launch_toast();
+    }
+    addTodo(event);
+  }
+}
+
+//delete a todo from the list
+function deleteTask(e) {
+  const item = e.target;
+
+  if (item.classList[0] === "trash-btn") {
+    const todo = item.parentElement;
+    todo.classList.add("fall");
+    removeLocalTodos(todo);
+    //delay the remove for animation
+    setTimeout(function () {
+      todo.remove();
+    }, TRASH_ANIMATION_TIMEOUT);
+  }
+}
+
+//clear all button clicked
+function clearAll(e) {
+  const item = e.target;
+  if (item.classList[0] === "clearAllBtn") {
+    todoList.classList.add("fall");
+
+     //delay the remove for animation
+    setTimeout(function () {
+      todoList.innerHTML = "";
+      localStorage.clear();
+      todoList.classList.remove("fall");
+      listStyling(0);
+    }, TRASH_ANIMATION_TIMEOUT);
+    pendingTasksCount.textContent = 0;
+    if (!todoInput.value.length) {
+      inputBox.classList.remove("inactive");
+      todoButton.classList.remove("inactive");
+    }
+    sortBtn.textContent = "Sort";
+  }
+}
+
+//return values from local storage as an array
+function checkLocalStorage() {
+  let todos;
+  if (localStorage.getItem("todos") === null) {
+    todos = [];
+  } else {
+    todos = JSON.parse(localStorage.getItem("todos"));
+  }
+  return todos;
+}
+
+//save data to local storage
+function saveLocalTodos(todo) {
+  let todos = checkLocalStorage();
+  todos.push(todo);
+  localStorage.setItem("todos", JSON.stringify(todos));
+
+  pendingTasksCount.textContent = todos.length;
+}
+
+//get and show the data from local storage after refreshing the page
+function getTodos() {
+  //is the storage empty?
+  let todos = checkLocalStorage();
+
+  todos.forEach(function (todo) {
+    const todoLi = document.createElement("li");
+    todoLi.classList.add("todo");
+    todoLi.innerText = todo;
+
+    //trash button
+    const trashButton = document.createElement("button");
+    trashButton.innerHTML = '<i  class="fas fa-trash"></i>';
+    trashButton.classList.add("trash-btn");
+    todoLi.appendChild(trashButton);
+    //append to list
+    todoList.appendChild(todoLi);
+    //Clear Todo Input value
+    todoInput.value = "";
+  });
```

Comment:
this chunk of code is duplicated here and in lines 241-255.
duplicated code should become a function that you can use multiple times